cmd/kubelet/app/server.go

```diff
@@ -605,9 +605,11 @@ func run(ctx context.Context, s *options.KubeletServer, kubeDeps *kubelet.Depend
 		klog.ErrorS(err, "Failed to register kubelet configuration with configz")
 	}
 
+	// Apply `ShowHiddenMetricsForVersion` separately for backward compatibility.
 	if len(s.ShowHiddenMetricsForVersion) > 0 {
 		metrics.SetShowHidden()
 	}
+	s.Metrics.Apply()
```

Comment:
I see unit tests coverage. I would suggest to write a simple e2e test under the e2e_node that validates that all the wiring works fine. There are a few examples that change kubelet config and `restartKubelet` to pick up the change.